common/middleware/set-services.js

```diff
@@ -0,0 +1,21 @@
+const services = require('../services')
+
+const setServices = (req, res, next) => {
+  const servicesCache = {}
+
+  const servicesWithReq = Object.keys(services).reduce((acc, serviceKey) => {
+    if (!servicesCache[serviceKey]) {
+      servicesCache[serviceKey] = services[serviceKey]
+    }
+
+    acc[serviceKey] = () => servicesCache[serviceKey]
```

Comment:
Could this be `acc[serviceKey] = servicesCache[serviceKey]` so that the services don't require instantiation in controllers/middleware?